frontend/packages/dev-console/src/components/add/layout/MasonryLayout.tsx

```diff
@@ -6,20 +6,33 @@ import './MasonryLayout.scss';
 type MasonryLayoutProps = {
   columnWidth: number;
   children: React.ReactElement[];
+  loading?: boolean;
+  LoadingComponent?: React.ComponentType<any>;
 };
 
-export const MasonryLayout: React.FC<MasonryLayoutProps> = ({ columnWidth, children }) => {
+export const MasonryLayout: React.FC<MasonryLayoutProps> = ({
+  columnWidth,
+  children,
+  loading,
+  LoadingComponent,
+}) => {
   const [width, setWidth] = React.useState<number>(0);
   const columnCount = React.useMemo(() => (width ? Math.floor(width / columnWidth) || 1 : null), [
     columnWidth,
     width,
   ]);
 
+  const loadingElements: React.ReactElement[] = LoadingComponent
+    ? Array.from({ length: columnCount }, (_, i) => <LoadingComponent key={i.toString()} />)
+    : [];
+
```

Comment:
It's not required to create the `LoadingComponent` when the `loading` prop is false. Maybe you can change it like this?

```suggestion
  const coumns: React.ReactElement[] =
    loading && LoadingComponent
      ? Array.from({ length: columnCount }, (_, i) => <LoadingComponent key={i.toString()} />)
      : children;
```